src/ucp/wireup/select.c

```diff
@@ -806,6 +807,32 @@ ucp_wireup_ep_init_flags(const ucp_wireup_select_params_t *select_params,
     return select_params->ep_init_flags | select_ctx->ucp_ep_init_flags;
 }
 
+static ucs_status_t
+ucp_wireup_add_cm_lane(const ucp_wireup_select_params_t *select_params,
+                       ucp_wireup_select_context_t *select_ctx)
+{
+    ucp_wireup_select_info_t select_info;
+
+    if (!(select_params->ep_init_flags & (UCP_EP_INIT_CM_WIREUP_CLIENT |
+                                          UCP_EP_INIT_CM_WIREUP_SERVER))) {
+        return UCS_OK;
+    }
+
+    select_info.reachable  = 1;  /**< UCT CM provides reachability by its API
```

Comment:
why do we need this field? we should not be adding lane desc which is not reachable
seems that latest change which introduced this field was not correct

Reply:
Maybe comment here and/or field name is not correct..
this field is needed to indicate that need to compare against prev value in selection logic to select the best transport

Reply:
see https://github.com/openucx/ucx/blob/10f09c95527fafe9943845651b4709dbc17e68b0/src/ucp/wireup/select.c#L252

Reply:
that location should use a local variable as it was before
select_info is the information to add to list of lanes. it should contain only reachable pairs of {rsc_index,remote_addr_index}

Reply:
- then need to have 2 flags: `is_reachable` as a local variable and `select_info.is_initialized` to do comparison with previous score.
 -  `ucp_wireup_update_select_info` is not called for unreachable pair for TL lanes
 - CM lane does not have valid {rsc_index,remote_addr_index} pair at all, select_info is filled just to have lane desc and reuse common flow for lanes creation.

maybe rename `select_info.reachable` to `select_info.is_initialized` would make it clearer?

Reply:
will be fixed in https://github.com/openucx/ucx/pull/4347